### PR TITLE
Add more type checking

### PR DESCRIPTION
**Pull request by szpak**

### CODE REVIEW

File: src/main/groovy/pl/wybcz/MicroserviceTemplateBuilder.groovy [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
+@CompileStatic
 class MicroserviceTemplateBuilder {
 
-    DslFactory dslFactory
-    PrBuilder prBuilder
-    List<GitProject> projects
+    private DslFactory dslFactory
+    private PrBuilder prBuilder
+    private List<GitProject> projects
```
Comment (szpak) [inline src/main/groovy/pl/wybcz/MicroserviceTemplateBuilder.groovy]: Maybe it is planned to set it directly via setter?

Comment (marcingrzejszczak) [inline src/main/groovy/pl/wybcz/MicroserviceTemplateBuilder.groovy]: yeah it was planned to make setting of all of that via maps - looks nicer from DSL perspective

**Comment by szpak** [comment]
Btw, it fails on Travis with:

```
java.io.IOException: Server returned HTTP response code: 403 for URL: https://api.github.com/orgs/microhackathon-2015-03-juglodz/repos
```

It works locally. Is there more tests run on Travis?

**Comment by marcingrzejszczak** [comment]
I guess it was a temporary problem - https://travis-ci.org/microservice-hackathon/jenkins/builds/56798328

**Comment by marcingrzejszczak** [comment]
BTW about the suggestion to make the fields `private`. I think I was already tired cause it actually seems like a good idea to make them private. We have additional methods that can set these values so there is no need to make the fields `public`.

**Comment by szpak** [comment]
Rebased to make them `private`.

**Comment by marcingrzejszczak** [comment]
Conflicts :/

**Comment by szpak** [comment]
After your recent changes I' not surprised :). I will check if there is still anything useful to apply.

**Comment by jakubnabrdalik** [comment]
Szpak, do you ever sleep at night?

**Comment by szpak** [comment]
Rebase the PR against current master.

**Comment by szpak** [comment]
Hmm, there could be problem with that PR. After merge deploying fails with:

```
Processing DSL script microservice-template-github.groovy
FATAL: startup failed:
file:/mnt/homeold/szpak/.jenkins-homes/.jenkins-hackathon3/jobs/pipeline-seed/workspace/src/main/groovy/pl/wybcz/pipeline/template/MicroserviceTemplateBuilder.groovy: 3: unable to resolve class groovy.transform.CompileStatic
 @ line 3, column 1.
   import groovy.transform.CompileStatic
   ^
```

It is possible that there is old Groovy used in Jenkins and/or in Job DSL plugin. Do you have similar problem?

**Comment by szpak** [comment]
Changes reverted as Jenkins uses Groovy 1.8.9...

**Pull request merged**